widget_driver_generator/lib/src/utils/package_info_provider.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:convert';
+import 'dart:io';
+
+class PackageInfoProvider {
+  final FileReader _fileReader;
+
+  PackageInfoProvider({
+    FileReader? fileReader,
+  }) : _fileReader = fileReader ?? FileReader();
+
+  /// Grabs the version string for widget_driver_generator from the pubspec.lock
+  Future<String> getPackageVersionString() async {
```

Comment:
You could use the yaml package to get the DOM of the lock file.
Like here https://github.com/bmw-tech/dart_apitool/blob/8002b6843679bcd72221f90e6e96e3ec04bbfadb/bin/main.dart#L31

Reply:
Thanks for the tip. 
Yeah that looks nice, but we wanted to avoid external dependencies when not super needed.
And here this logic is not crucial for the functionality of the widget_driver framework.. just to print the version number out.
And our "brute-force" approach does the trick ;) 

So it might be worth it having our code with less dependencies..
And we do unit test the code :)